config/m4/compiler.m4

```diff
@@ -186,22 +186,44 @@ AC_DEFUN([CHECK_COMPILER_FLAG],
                 AC_MSG_CHECKING([compiler flag -diag-error 10006])
                 SAVE_CFLAGS="$CFLAGS"
                 CFLAGS="$BASE_CFLAGS $CFLAGS -diag-error 10006"
+                AC_LANG_PUSH([C])
                 AC_COMPILE_IFELSE([AC_LANG_SOURCE([[int main(){return 0;}]])],
                                   [BASE_CFLAGS="$BASE_CFLAGS -diag-error 10006"
+                                   BASE_CXXFLAGS="$BASE_CXXFLAGS -diag-error 10006"
                                    AC_MSG_RESULT([yes])],
                                   [AC_MSG_RESULT([no])])
+                AC_LANG_POP([C])
                 CFLAGS="$SAVE_CFLAGS"
                ],
                [])
+])
+
+#
+# CHECK_COMPILER_FLAG
+# Usage: CHECK_COMPILER_FLAG([name], [flag], [program], [if-true], [if-false])
+#
+# The macro checks if program may be compiled using specified flag
+#
+AC_DEFUN([CHECK_COMPILER_FLAG],
+[
+#
+# Force ICC treat command line warnings as errors.
+# This evaluation should be called prior to all other compiler flags evals
+#
```

Comment:
can remove this comment, it already appears above

Reply:
removed